plugins/inputs/redfish/README.md

```diff
@@ -0,0 +1,165 @@
+# Redfish Input Plugin
+
+The `redfish` plugin gathers  metrics and status information about CPU temperature, fanspeed, Powersupply, voltage, hostname and Location details(datacenter,placement,rack and room) of Dell hardware servers for which redfish is enabled.
+And also metrics like CPU temperature,fanspeed, Powersupply and hostname metrics for HP Hardware server(redfish should be enabled).
+
+Note: Currently this plugin only supports DELL and HP servers.
+
+
+
+### Configuration
+
+```toml
+[[inputs.redfish]]
+## Server OOB-IP
+host = "http://192.0.0.1"
+
+## Username,Password for hardware server
+basicauthusername = "test"
+basicauthpassword = "test"
+## Server Vendor(dell or hp)
+server= "dell"
+## Resource Id for redfish APIs
+id="System.Embedded.1"
+## Optional TLS Config
+# tls_ca = "/etc/telegraf/ca.pem"
+# tls_cert = "/etc/telegraf/cert.pem"
+# tls_key = "/etc/telegraf/key.pem"
+## Use TLS but skip chain & host verification
+# insecure_skip_verify = false
+
+## Amount of time allowed to complete the HTTP request
+# timeout = "5s"
+```
+
+### Metrics for Dell Servers
+
+- cpu_temperatures
+	- tags:
+		- hostname
+		- name
+		- oob_ip
+		- host
+	- Fields:
+		- datacenter
+		- temperature
+		- health
+		- rack
+		- room
+		- row
+		- state
```

Comment:
Great breakdown on the measurements, thanks a lot for the suggestion, I think it would be appropriate if health and state will also be sent as tags as they are string values.

Reply:
Tags do a few things:
- they uniquely identify a record (as a group),
- they are indexed so if you expect people to search by them it can be more performant to include them as tags
- they must be strings
- all the possible different tag values are indexed, so if a field has a lot of different possible values, it's not a good candidate for a tag.

some examples: types and enums are probably good tags. Percentages, or arbitrary strings are not.

based on that, you probably have enough info to decide which of these fields should be tags.